packages/block-editor/src/store/selectors.js

```diff
@@ -1618,25 +1618,32 @@ export const getInserterItems = createSelector(
 				blockVariations.push( ...variations.map( variationMapper ) );
 			}
 		}
-		// Prioritize core blocks's display in inserter.
-		const prioritizeCoreBlocks = ( a, b ) => {
-			const coreBlockNamePrefix = 'core/';
-			const firstIsCoreBlock = a.name.startsWith( coreBlockNamePrefix );
-			const secondIsCoreBlock = b.name.startsWith( coreBlockNamePrefix );
-			if ( firstIsCoreBlock && secondIsCoreBlock ) {
-				return 0;
-			}
-			return firstIsCoreBlock && ! secondIsCoreBlock ? -1 : 1;
-		};
 		// Ensure core blocks are prioritized in the returned results,
 		// because third party blocks can be registered earlier than
 		// the core blocks (usually by using the `init` action),
 		// thus affecting the display order.
 		// We don't sort reusable blocks as they are handled differently.
+		const toTyped = ( blocks, block ) => {
```

Comment:
Nitpick: renaming this might better communicate the intent. Feel free to ignore this comment, however. 
```suggestion
		const groupByType = ( blocks, block ) => {
```

Reply:
Thanks for the suggestion @dcalhoun 🙇 , I've just applied it in [this commit](https://github.com/WordPress/gutenberg/pull/34078/commits/a72b5abcfc9694d27005a824a757a02feedf016a).

Reply:
I named it `toTyped` not because I particularly liked that name, but because I like for array operations to read like prose as much as possible, e.g., `blockVariations.reduce( toTyped )`. I just couldn't think of a good name when I put up the draft PR. The name wasn't the focus at the time, but rather the nature of the solution.

I don't feel particularly strongly about changing it again, but maybe something like `blockVariations.reduce( toGroupedBlockTypes )` would be more expressive?

Also, I believe `groupByType` can be moved to the module lexical scope instead of the local functional scope because I don't believe I closured anything in the function. I intended to do that in the draft PR and forgot.

Reply:
> [...] I like for array operations to read like prose as much as possible, e.g., `blockVariations.reduce( toTyped )`. 

> I don't feel particularly strongly about changing it again, but maybe something like `blockVariations.reduce( toGroupedBlockTypes )` would be more expressive?

Ah, I see. I did not read or recognize that. That does make sense and a name like `toGroupedBlockTypes` could express that well. I'm fine with whichever name @fluiddot prefers. 👍🏻

Reply:
I don't have a strong opinion on the naming either but since this PR should be merged as soon as possible as we're aiming to include it in the next release, I'm leaning towards preventing more changes.

> Also, I believe groupByType can be moved to the module lexical scope instead of the local functional scope because I don't believe I closured anything in the function. I intended to do that in the draft PR and forgot.

I think it's ok to keep it there, the [previous function `prioritizeCoreBlocks`](https://github.com/WordPress/gutenberg/blob/20d9c54f6684627486c9e59a4a24848b4bd4f4ef/packages/block-editor/src/store/selectors.js#L1621-L1630) was defined in the same place and its use is mostly exclusive to the block type sorting logic.